functions.php

```diff
@@ -1299,63 +1300,60 @@ function wp_autoupdates_toggle_auto_updates() {
 		wp_send_json_error( array( 'error' => __( 'Invalid data. No selected item.', 'wp-autoupdates' ) ) );
 	}
 
-	$type  = sanitize_text_field( $_POST['type'] );
```

Comment:
I believe the sanitizing was at the request of @whyisjake .  If he's OK with these no longer being sanitized, then rest looks good to me.

@azaozz I just pushed an additional commit with 2 other minor changes, so be sure to check those.

Reply:
Right. All strings coming from the user, (`$_GET`, `$_POST`, `$_REQUEST`, most from `$_SERVER`, etc.) must be sanitized. However strings that must match a particular, limited example can be matched directly. In this case `$_POST['state']` can only be `enable` or `disable`, all other strings there are ignored. Then a direct match makes more sense as it's faster.

Similarly when a number is expected, it would be enough to cast or int (or do `intval()` (with some limitations), instead of first sanitizing the string.